Code-Code/TypePrediction-TypeScript/dataset/process_datafiles.py

```diff
@@ -0,0 +1,161 @@
+    print('Wrote {} records to {}'.format(len(data), output_path))
+
+
+def load_jsonl(input_path) -> list:
+    """
+    Read list of objects from a JSON lines file.
+    """
+    data = []
+    with open(input_path, 'r', encoding='utf-8') as f:
+        for line in f:
+            data.append(json.loads(line.rstrip('\n|\r')))
+    print('Loaded {} records from {}'.format(len(data), input_path))
+    return data
+
+def calculate_vocab(all_training_labels, vocab_size):
+    c = Counter(all_training_labels)
+    vocab = ['UNK']+list(list(zip(*c.most_common(len(c))))[0])
+    tag2id = {tag: id for id, tag in enumerate( vocab, 0)}
+    id2tag = {id: tag for tag, id in tag2id.items()}
+    truncated_vocab = list(tag2id.keys())[:vocab_size+1]
+    with open('vocab_{}.txt'.format(int(vocab_size)), 'w') as filehandle:
+        for listitem in truncated_vocab:
+            filehandle.write('%s\n' % listitem)
+    return truncated_vocab
+
+#../vocab_50k.txt
+def load_vocab(vocab_file):
+    with open(vocab_file, 'r') as file:
+        vocab = file.readlines()
+        vocab = [line.rstrip() for line in vocab]
+        return vocab
+
+
+
+def create_dataframe(datafile, vocabulary, test_or_valid = False):
+    data_for_frame = []
+    input_ids = []
+    labels_ids = []
+    for row in datafile:
+        for f_name, file in row['filedata'].items():
+            if not file['annotations']:
+                continue
+            simplified_annotations = [annot['ty'] for annot in file['annotations']]
+
+            for i in range(len(simplified_annotations)):
+                simplified_annotations[i]['category']=file['annotations'][i]['category'] 
+            if test_or_valid:
+                label_map = {annotation['loc']:annotation['type'] for annotation in simplified_annotations if annotation['category'] == 'UserAnnot'}
+            else:
+                label_map = {annotation['loc']:annotation['type'] for annotation in simplified_annotations}
+            labels = [label_map[i] if i in label_map else None for i in range(len(file['tokens']))]
+            n_labels = []
+            for label in labels:
+                n_label =label
+                if label is None:
+                    n_labels.append(n_label)
+                    continue
+               
+                if label not in vocabulary:
+                    n_label = 'UNK'
+                elif label =='any':
+                    n_label = None
+                n_labels.append(n_label)
+        
+            if all(v is None for v in n_labels):
+                continue
+            
+            data_row = {'tokens': file['tokens'], 'labels': n_labels, 'url':row['url'], 'path':f_name, 'commit_hash':row['commit_hash'], 'file':os.path.basename(f_name)}
+            data_for_frame.append(data_row)
+    return data_for_frame
+
+def main(vocab_size):
+
+    train_datafile0 = load_jsonl('ManyTypes4TypeScript_zenodo/datafiles/splits/train_datafile0.jsonl')
```

Comment:
It seems there's no `splits` in this path

Reply:
Link in README.txt does not download the latest version for some reason. Adding an argument to link to download the latest version with the splits file structure. Will be in new commit.

Reply:
`wget https://zenodo.org/record/6387001/files/ManyTypes4TypeScript.tar.gz?download=1 -O ManyTypes4TypeScript.tar.gz`

downloads the version with splits, it is in the latest commit.